tools/webhook-apicoverage/resourcetree/resourceforest.go

```diff
@@ -31,6 +31,16 @@ type ResourceForest struct {
 	ConnectedNodes map[string]*list.List // Head of the linked list keyed by nodeData.fieldType.pkg + nodeData.fieldType.Name()
 }
 
+// AddResourceTree adds a resource tree to the resource forest.
+func (r *ResourceForest) AddResourceTree(resourceName string, resourceType reflect.Type) {
+	tree := ResourceTree{
+		ResourceName: resourceName,
+		Forest: r,
```

Comment:
Is it unavoidable to for r and tree to inter-referencing each other? This could make adding/moving/deleting harder

Reply:
It is because of connected nodes. Its a reverse reference back to logical parent, so we would not be modification will only be from forest to tree.